packages/geo/src/lib/draw/shared/draw-style.service.ts

```diff
@@ -10,160 +11,176 @@ import { MapService } from '../../map/shared/map.service';
   })
 export class DrawStyleService {
 
-    private fillColor: string = 'rgba(255,255,255,0.4)';
-    private strokeColor: string = 'rgba(51,153,204,1)';
-    private drawCounter: number = 1;
-    private toggleLabel = true;
-    private icon;
-
-    constructor(
-      private mapService: MapService
-    ) {}
-
-    getFill(): string {
-        return this.fillColor;
-    }
-
-    setFill(fillColor: string) {
-        this.fillColor = fillColor;
-    }
-
-    getStroke(): string {
-        return this.strokeColor;
+  private fillColor: OlColor = 'rgba(255,255,255,0.4)';
+  private strokeColor: OlColor = 'rgba(143,7,7,1)';
+  private strokeWidth: number = 1;
+  private drawCount: number = 1;
+  private labelsAreShown = true;
+  private icon: string;
+
+  constructor(
+    private mapService: MapService
+  ) {}
+
+  getFillColor(): OlColor {
+    return this.fillColor;
+  }
+
+  setFillColor(fillColor: OlColor) {
+    this.fillColor = fillColor;
+  }
+
+  getStrokeColor(): OlColor {
+    return this.strokeColor;
+  }
+
+  setStrokeColor(strokeColor: OlColor) {
+    this.strokeColor = strokeColor;
+  }
+
+  getStrokeWidth(): number {
+    return this.strokeWidth;
+  }
+
+  setStrokeWidth(strokeWidth: number) {
+    this.strokeWidth = strokeWidth;
+  }
+
+  getDrawCount() {
+    return this.drawCount;
+  }
+
+  getLabelsAreShown() {
+    return this.labelsAreShown;
+  }
+
+  toggleLabelsAreShown() {
+    this.labelsAreShown = !this.labelsAreShown;
+  }
+
+  raiseDrawCount() {
+    this.drawCount += 1;
+  }
+
+  setIcon(icon: string) {
+    this.icon = icon;
+  }
+
+  getIcon() {
+    return this.icon;
+  }
+
+  createDrawingLayerStyle(feature, resolution, labelsAreShown?: boolean, icon?: string): OlStyle.Style {
+    let style;
+    let labelsAreOffset: boolean = false;
+    const proj = this.mapService.getMap().ol.getView().getProjection().getCode();
```

Comment:
maybe refer to the view controller.
this.mapService.getMap().viewController.getOlProjection().getCode()

Reply:
or getMap().projection

Reply:
![image](https://user-images.githubusercontent.com/7397743/129791885-7f05bff1-3bcd-4e8a-a275-11cd58e20b2f.png)